integration/shareddata/shareddata.go

```diff
@@ -197,7 +199,43 @@ func (values *Values[idType]) IsCompatible(backend string) bool {
 	return slices.Contains(values.backends, backend)
 }
 
+// BenchmarkProvider is implemented by shared data sets that provide documents for benchmarks.
+type BenchmarkProvider interface {
+	// Hash returns expected hash of all provider documents.
+	Hash() string
+
+	// Docs returns iterator that returns all documents from provider.
+	// They should be always in deterministic order.
+	Docs() iterator.Interface[struct{}, bson.D]
+}
```

Comment:
We introduce another provider, as from `Provider` we require data to be stored in random order, and `BenchmarkProvider` needs to be deterministic.

We use iterator in place of slice, as benchmarks might include much more data than tests, and it shouldn't be store in memory.

Reply:
Does `Provider` needs to be random order? I think `Provider` being random is due to design or implementation. Because if provider can be deterministic, it's easier to maintain one implementation than having multiple.

Reply:
`Provider` for integration tests returns documents in random order by design – to spot problems in our tests that depend on some specific order.

Let's keep those interfaces separate for now. We might merge them later.

Reply:
true, the randomness can make tests more robust :+1: